### PR TITLE
feat(ux): Warn on unsaved settings navigation

### LINKED ISSUE #42: [UX] Add unsaved changes warning on settings navigation
## Problem

Settings page tracks `hasChanges` but doesn't warn user on navigation:

```typescript
// app/dashboard/settings/page.tsx:80-86
const hasChanges = settings && (
  emailNotifications !== settings.emailNotifications ||
  notifyOnDown !== settings.notifyOnDown ||
  // ...
);
```

User makes changes, accidentally clicks "Back to Dashboard", loses all changes with no warning.

## Location

`app/dashboard/settings/page.tsx`

## Proposed Fix

Add `beforeunload` handler:
```typescript
useEffect(() => {
  if (hasChanges) {
    const handler = (e: BeforeUnloadEvent) => {
      e.preventDefault();
      e.returnValue = '';
    };
    window.addEventListener('beforeunload', handler);
    return () => window.removeEventListener('beforeunload', handler);
  }
}, [hasChanges]);
```

---
🤖 Created by `/groom`

### PR DESCRIPTION
**Pull request by phrazzld**
## Summary

Closes #42.

Users who made changes to notification settings could lose their work by accidentally navigating away — clicking "Back to Dashboard" or closing the tab with no warning. This PR adds a browser-native `beforeunload` warning and extracts it into a reusable hook.

## Changes

- **`hooks/useUnsavedChangesWarning.ts`** — New hook: attaches/detaches a `beforeunload` listener based on a boolean `hasChanges` flag. Extracted to keep the settings component clean and make this reusable elsewhere.
- **`app/dashboard/settings/page.tsx`** — One-line integration: `useUnsavedChangesWarning(!!hasChanges)` after the existing `hasChanges` derivation.
- **`hooks/__tests__/useUnsavedChangesWarning.test.ts`** — 5 tests covering: no listener when `hasChanges=false`, listener added when `true`, cleanup on unmount, toggle off removes listener without re-adding, and handler calls `preventDefault`.

## Acceptance Criteria

- [x] User sees browser's native "Leave site?" dialog when navigating away with unsaved changes
- [x] No warning appears when there are no unsaved changes
- [x] Warning clears after saving

## Manual QA

1. `pnpm dev`
2. Navigate to **Dashboard → Settings**
3. Change any setting (e.g. toggle Email Notifications)
4. Click "Back to Dashboard" — browser should show a "Leave site?" confirmation dialog
5. Dismiss the dialog — you stay on settings
6. Save the settings, then navigate away — no dialog should appear
7. Reload the page with unsaved changes — dialog should appear

## Before / After

**Before:** User edits notification settings, clicks "Back to Dashboard", all changes silently discarded.

**After:** Browser-native "Leave site?" dialog intercepts the navigation. User can choose to stay and save, or confirm they want to leave.

No visual change to the settings UI itself — this is a browser-level guard.

## Test Coverage

- `hooks/__tests__/useUnsavedChangesWarning.test.ts` — full coverage of the hook
  - `does not add beforeunload listener when hasChanges is false`
  - `adds beforeunload listener when hasChanges is true`
  - `removes beforeunload listener on cleanup when hasChanges is true`
  - `removes listener and does not re-add when hasChanges changes from true to false`
  - `calls event.preventDefault on beforeunload event`

**Gap:** No integration test for `SettingsPage` consuming the hook — the hook itself is thoroughly unit-tested, and the integration is a single line, so the cost of a full settings page render test is higher than the risk.

## Summary by CodeRabbit

* **New Features**
  * Added unsaved-changes protection: prompts users before navigating away or closing the tab when there are unsaved modifications.
  * In-app confirmation modal with Stay/Leave actions, Escape and backdrop-to-cancel handling, and interception of internal link navigation.

* **Tests**
  * Added comprehensive tests covering browser unload and in-app interception, modifier-key/middle-click bypasses, confirm/cancel flows, listener cleanup, and state transitions.

---

## Fix Notes (post-review)

**Before (7ed3f13):** `beforeunload` guarded browser unload only. Clicking "Back to Dashboard" or "Billing" tab — in-app Next.js `<Link>` navigation — silently discarded unsaved changes. Only 2 of 5 navigation paths were guarded (APOLLO finding: DashboardNavbar logo, billing icon, settings icon, and Footer Dashboard link were all unguarded).

**After (2cca874):** `useUnsavedChangesWarning` installs a document-level capture-phase click listener when `hasChanges` is true. This fires before any `<Link>` onClick, intercepts all internal anchor navigations across the entire page (including shared layout components), and sets `pendingNavigation` state. settings/page.tsx renders a confirmation modal driven entirely by the hook. All 5 navigation paths are now guarded.

Hook API changed from `void` → `{ pendingNavigation, confirmNavigation, cancelNavigation }`. Tests expanded from 5 → 14.

---

## Polish Pass

**Before:** Working feature with a UX gap (modifier-key clicks silently blocked), no Escape key handling on the modal, no ARIA attributes, fragile state cleanup in confirmNavigation, and an inline modal diverging from established codebase patterns.

**After:**
- **Bug fix:** Cmd/Ctrl/Shift/Alt+click and middle-click now bypass the guard (opening in new tab doesn't abandon the page)
- **State hygiene:** confirmNavigation clears pendingNavigation eagerly instead of relying on the navigation→hasChanges→cleanup chain
- **Accessibility:** Escape key dismisses modal (matching DeleteConfirmModal), role=dialog, aria-modal, aria-labelledby/describedby
- **Structure:** Inline modal extracted to UnsavedChangesModal component within the same file
- **Tests:** 4 new edge-case tests (modifier keys, eager state clearing). Total: 19 hook tests
- **Follow-up:** #99 filed for shared ConfirmModal extraction (2 modal instances with identical structure)

All quality gates pass: typecheck, lint, 936 tests.

### CODE REVIEW

File: hooks/__tests__/useUnsavedChangesWarning.test.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
+import { describe, test, expect, vi, beforeEach, afterEach } from "vitest";
+import { renderHook, act } from "@testing-library/react";
+import { useUnsavedChangesWarning } from "../useUnsavedChangesWarning";
```
[the remark this answers is not in the document]
Comment (phrazzld) [reply hooks/__tests__/useUnsavedChangesWarning.test.ts]: Fixed in 7ed3f13: renamed to `useUnsavedChangesWarning.test.tsx` and updated the import to `@/hooks/useUnsavedChangesWarning`.

**Comment by phrazzld** [comment]
## PR Unblocked (Round 2)

**Conflicts**: none  
**CI**: was failing — APOLLO (correctness) correctly identified that the initial fix guarded only 2 of 5 navigation paths  
**Reviews addressed**:
- **Codex P1** (7ed3f13): added `handleLinkClick` on the 2 in-page `<Link>` components — partially fixed
- **APOLLO correctness** (2cca874): moved guard into `useUnsavedChangesWarning` as a document-level capture-phase listener — now guards DashboardNavbar (logo, billing icon, settings icon) and Footer (Dashboard link) which are outside the settings page's component tree
- **CodeRabbit minor** (7ed3f13): renamed test file to `.test.tsx`, updated import to `@/hooks/...`

**Architecture change**: hook now returns `{ pendingNavigation, confirmNavigation, cancelNavigation }` instead of `void`. settings/page.tsx renders the confirmation modal driven by hook state — no per-link handlers needed.

**Tests**: 5 → 14 (added click interceptor, state machine, and cleanup path coverage)

Ready for re-review.

File: app/dashboard/settings/page.tsx [hunk 2]
```diff
@@ -363,6 +385,43 @@ export default function SettingsPage() {
...
+        >
+          <div className="relative w-full max-w-sm mx-4 bg-[var(--color-bg-primary)] shadow-[var(--shadow-lg)] border border-[var(--color-border-subtle)] rounded-[var(--radius-lg)]">
+            <div className="px-6 py-6">
+              <h2 className="font-display text-xl text-[var(--color-text-primary)] mb-3">
+                Unsaved Changes
+              </h2>
+              <p className="text-sm text-[var(--color-text-secondary)] leading-relaxed mb-6">
+                You have unsaved changes. If you leave now, they will be lost.
+              </p>
+              <div className="flex gap-3 justify-end">
```
[the remark this answers is not in the document]
Comment (phrazzld) [reply app/dashboard/settings/page.tsx]: Fixed in 3060e5c: replaced `font-display` with `font-serif` on the `<h2>` and added `font-sans` to the `<p>` per coding guidelines.

**Comment by phrazzld** [comment]
## PR Unblocked

**Conflicts**: none

**CI**: green (APOLLO: 3 iterations to resolve; final fix in 4bfc3d7)

**Reviews addressed**:

| Thread | Finding | Fix | Commit |
|--------|---------|-----|--------|
| Codex P1 | `beforeunload` only fires on browser unload — Next.js `<Link>` client-side navigation was unguarded | Redesigned hook to use a document capture-phase click interceptor (`document.addEventListener("click", fn, true)`), intercepting all internal `a[href]` clicks before React's event system. Hook now returns `{ pendingNavigation, confirmNavigation, cancelNavigation }`. Settings page wires modal to hook state. 9 new tests added. | 2cca874 |
| CodeRabbit (import alias + file ext) | Test file used relative import and `.test.ts` extension | Updated to `@/hooks/useUnsavedChangesWarning`, renamed to `.test.tsx` | 7ed3f13 |
| APOLLO (tel: / other protocols) | Allowlist let `javascript:`, `tel:`, `data:` etc. through to `router.push()` | Replaced allowlist with positive check: only intercept hrefs starting with `/`. Added `tel:` passthrough test. | 4bfc3d7 |
| CodeRabbit (font classes) | `font-display` on modal heading; missing `font-sans` on body text | `font-display` → `font-serif`, added `font-sans` to `<p>` | 3060e5c |

Ready for re-review.
